client/k8sclient/k8sclient.go

```diff
@@ -62,60 +62,52 @@ func DefaultConfig() Config {
 
 // New returns a Kubernetes Clientset with the provided configuration.
 func New(config Config) (kubernetes.Interface, error) {
-	rawClientConfig, err := getRawClientConfig(config)
+	restConfig, err := toClientGoRESTConfig(config)
 	if err != nil {
 		return nil, err
 	}
 
-	client, err := kubernetes.NewForConfig(rawClientConfig)
+	client, err := kubernetes.NewForConfig(restConfig)
 	if err != nil {
 		return nil, err
 	}
 
 	return client, nil
 }
 
-func newRawClientConfig(config Config) *rest.Config {
+func toClientGoRESTConfig(config Config) (*rest.Config, error) {
+	if config.InCluster {
+		config.Logger.Log("debug", "creating in-cluster config")
+		c, err := rest.InClusterConfig()
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		return c, nil
+	}
```

Comment:
With this the `MaxQPS` and `MaxBurst` are not configured when the in cluster client is used. What about just copy/paste from the crdclient? I fixed this there.

Reply:
Done.